go/appencryption/README.md

```diff
@@ -0,0 +1,257 @@
+# Asherah - Go
+Application level envelope encryption SDK for Golang with support for cloud-agnostic data storage and key management.
+
+[![GoDoc](https://godoc.org/github.com/godaddy/asherah/go/appencryption?status.svg)](https://godoc.org/github.com/godaddy/asherah/go/appencryption)
+
+  * [Quick Start](#quick-start)
+  * [How to Use Asherah](#how-to-use-asherah)
+    * [Define the Metastore](#define-the-metastore)
+    * [Define the Key Management Service](#define-the-key-management-service)
+    * [Define the Crypto Policy](#define-the-crypto-policy)
+    * [(Optional) Enable Metrics](#optional-enable-metrics)
+    * [Build a Session Factory](#build-a-session-factory)
+    * [Performing Cryptographic Operations](#performing-cryptographic-operations)
+  * [Documentation](#documentation)
+  * [Development Notes](#development-notes)
+
+## Quick Start
+
+```go
+crypto := aead.NewAES256GCM()
+// Create a session factory. The builder steps used below are for testing only.
+factory := appencryption.NewSessionFactory(
+    &appencryption.Config{ 
+        Service: "reference_app",
+        Product: "productId",
+        Policy:  appencryption.NewCryptoPolicy(),
+    },
+    persistence.NewMemoryMetastore(),
+    kms.NewStatic("mysupersecretstaticmasterkey!!!!", crypto)
+    crypto,
+)
+// Now create a cryptographic session for a partition.
+sess, err := factory.GetSession("shopper123")
+if err != nil {
+    panic(err)
+}
+// Close frees the memory held by the intermediate keys used in this session
+defer sess.Close() 
+	
+ // Now encrypt some data
+ dataRow, err := sess.Encrypt([]byte("mysupersecretpayload"))
+if err != nil {
+	panic(err)
+}
+
+//Decrypt the data
+data, err := sess.Decrypt(*dataRow)
+if err != nil {
+	panic(err)
+}
+```
+
+A more extensive example is the [Reference Application](cmd/example/README.md), which will evolve along 
```

Comment:
Is `cmd/example` just a placeholder until `go/referenceapp` is merged?

Reply:
yes, just didn't want to insert a invalid link here.